app/cdap/components/Metadata/SearchResults/helper.ts

```diff
@@ -0,0 +1,263 @@
+    foundIn.push(METADATA_FILTERS[2]);
+    metadataFiltersList[2].count++;
+  }
+  if (systemTags.search(regex) > -1) {
+    foundIn.push(METADATA_FILTERS[3]);
+    metadataFiltersList[3].count++;
+  }
+
+  // Properties
+  let userProperties = entityObj.metadata.properties.filter(
+    (property: IProperty) => property.scope === 'USER'
+  );
+  userProperties = JSON.stringify(convertToObject(userProperties));
+  let systemProperties = entityObj.metadata.properties.filter(
+    (property: IProperty) => property.scope === 'SYSTEM' && property.name !== 'schema'
+  );
+  systemProperties = JSON.stringify(convertToObject(systemProperties));
+  if (userProperties.search(regex) > -1) {
+    foundIn.push(METADATA_FILTERS[4]);
+    metadataFiltersList[4].count++;
+  }
+  if (systemProperties.search(regex) > -1) {
+    foundIn.push(METADATA_FILTERS[5]);
+    metadataFiltersList[5].count++;
+  }
+
+  // Schema
+  const schema = entityObj.metadata.properties.find(
+    (property: IProperty) => property.name === 'schema' && property.scope === 'SYSTEM'
+  );
+  if (schema && schema.value.search(regex) > -1) {
+    foundIn.push(METADATA_FILTERS[6]);
+    metadataFiltersList[6].count++;
+  }
+
+  return foundIn;
+}
+
+function filterResults(
+  fullResults: ISearchResult[],
+  entityFilters: IFilterItem[],
+  metadataFilters: IFilterItem[]
+) {
+  const filter = [];
+  entityFilters.forEach((entity) => {
+    if (entity.isActive) {
+      filter.push(entity.filter);
+    }
+  });
+
+  const entitySearchResults = fullResults.filter((result) => {
+    return filter.indexOf(result.type) > -1 ? true : false;
+  });
+
+  const metadataFilter = [];
+  metadataFilters.forEach((metadata) => {
+    if (metadata.isActive) {
+      metadataFilter.push(metadata.name);
+    }
+  });
+
+  return entitySearchResults.filter((result) => {
+    if (result.queryFound.length === 0) {
+      return true;
+    }
+    return intersection(metadataFilter, result.queryFound).length > 0;
+  });
+}
+
+export function applyFilter(
+  operation: Operations,
+  filter: IFilterItem,
+  isMetadata: boolean,
+  results: ISearchResult[],
+  entityFilters: IFilterItem[],
+  metadataFilters: IFilterItem[]
+) {
+  const filterObj = isMetadata ? metadataFilters : entityFilters;
+  switch (operation) {
+    case Operations.All:
+      filterObj.forEach((entity) => {
+        entity.isActive = true;
+      });
+      break;
+    case Operations.Only:
+      filterObj.forEach((entity) => {
+        entity.isActive = entity.name === filter.name;
+      });
+      break;
+    case Operations.Toggle:
+      const entityToToggle = filterObj.find((entity) => entity.name === filter.name);
+      entityToToggle.isActive = !entityToToggle.isActive;
+      break;
+    default:
+  }
+  return {
+    results: filterResults(results, entityFilters, metadataFilters),
+    entityFilters,
+    metadataFilters,
+  };
+}
+
+export function getResultCount(currentPage: number, total: number) {
```

Comment:
I think we want to jsdoc most exported funcitons right?

Reply:
Done